swarm/network/kademlia_test.go

```diff
@@ -89,65 +99,300 @@ func TestNeighbourhoodDepth(t *testing.T) {
 
 	baseAddress := pot.NewAddressFromBytes(baseAddressBytes)
 
-	closerAddress := pot.RandomAddressAt(baseAddress, 7)
-	closerPeer := newTestDiscoveryPeer(closerAddress, kad)
-	kad.On(closerPeer)
+	// generate the peers
+	var peers []*Peer
+	for i := 0; i < 7; i++ {
+		addr := pot.RandomAddressAt(baseAddress, i)
+		peers = append(peers, newTestDiscoveryPeer(addr, kad))
+	}
+	var sevenPeers []*Peer
+	for i := 0; i < 2; i++ {
+		addr := pot.RandomAddressAt(baseAddress, 7)
+		sevenPeers = append(sevenPeers, newTestDiscoveryPeer(addr, kad))
+	}
+
+	testNum := 0
+	// first try with empty kademlia
 	depth := kad.NeighbourhoodDepth()
 	if depth != 0 {
-		t.Fatalf("expected depth 0, was %d", depth)
+		t.Fatalf("%d expected depth 0, was %d", testNum, depth)
 	}
+	testNum++
 
-	sameAddress := pot.RandomAddressAt(baseAddress, 7)
-	samePeer := newTestDiscoveryPeer(sameAddress, kad)
-	kad.On(samePeer)
+	// add one peer on 7
+	kad.On(sevenPeers[0])
 	depth = kad.NeighbourhoodDepth()
 	if depth != 0 {
-		t.Fatalf("expected depth 0, was %d", depth)
+		t.Fatalf("%d expected depth 0, was %d", testNum, depth)
 	}
+	testNum++
 
-	midAddress := pot.RandomAddressAt(baseAddress, 4)
-	midPeer := newTestDiscoveryPeer(midAddress, kad)
-	kad.On(midPeer)
+	// add a second on 7
+	kad.On(sevenPeers[1])
 	depth = kad.NeighbourhoodDepth()
-	if depth != 5 {
-		t.Fatalf("expected depth 5, was %d", depth)
+	if depth != 0 {
+		t.Fatalf("%d expected depth 0, was %d", testNum, depth)
+	}
+	testNum++
+
+	// add from 0 to 6
+	for i, p := range peers {
+		kad.On(p)
+		depth = kad.NeighbourhoodDepth()
+		if depth != i+1 {
+			t.Fatalf("%d.%d expected depth %d, was %d", i+1, testNum, i, depth)
+		}
 	}
+	testNum++
 
-	kad.Off(midPeer)
+	kad.Off(sevenPeers[1])
 	depth = kad.NeighbourhoodDepth()
-	if depth != 0 {
-		t.Fatalf("expected depth 0, was %d", depth)
+	if depth != 6 {
+		t.Fatalf("%d expected depth 6, was %d", testNum, depth)
 	}
+	testNum++
 
-	fartherAddress := pot.RandomAddressAt(baseAddress, 1)
-	fartherPeer := newTestDiscoveryPeer(fartherAddress, kad)
-	kad.On(fartherPeer)
+	kad.Off(peers[4])
 	depth = kad.NeighbourhoodDepth()
-	if depth != 2 {
-		t.Fatalf("expected depth 2, was %d", depth)
+	if depth != 4 {
+		t.Fatalf("%d expected depth 4, was %d", testNum, depth)
 	}
+	testNum++
 
-	midSameAddress := pot.RandomAddressAt(baseAddress, 4)
-	midSamePeer := newTestDiscoveryPeer(midSameAddress, kad)
-	kad.Off(closerPeer)
-	kad.On(midPeer)
-	kad.On(midSamePeer)
+	kad.Off(peers[3])
 	depth = kad.NeighbourhoodDepth()
-	if depth != 2 {
-		t.Fatalf("expected depth 2, was %d", depth)
+	if depth != 3 {
+		t.Fatalf("%d expected depth 3, was %d", testNum, depth)
 	}
+	testNum++
+}
 
-	kad.Off(fartherPeer)
-	log.Trace(kad.string())
-	time.Sleep(time.Millisecond)
-	depth = kad.NeighbourhoodDepth()
-	if depth != 0 {
-		t.Fatalf("expected depth 0, was %d", depth)
+func TestHealthSimple(t *testing.T) {
```

Comment:
Yes, because to evaluate `Healthy` you need a `PeerPot`, which is only available if you're in god mode. A node in the network does not have this knowledge.